dotCMS/src/main/java/com/dotmarketing/startup/runonce/Task04225UpdateSystemFolderIdentifier.java

```diff
@@ -0,0 +1,110 @@
+    public static final String DROP_CONSTRAINT_QUERY = "ALTER TABLE Folder drop constraint folder_identifier_fk";
+    public static final String MYSQL_DROP_CONSTRAINT_QUERY = "ALTER TABLE Folder DROP FOREIGN KEY folder_identifier_fk";
+    public static final String CREATE_CONSTRAINT_QUERY = "ALTER TABLE Folder add constraint folder_identifier_fk foreign key (identifier) references identifier(id)";
+
+    @Override
+    public boolean forceRun() {
+        return true;
+    }
+
+    @Override
+    public void executeUpgrade() throws DotDataException {
+        DotConnect dc = new DotConnect();
+
+        //Drop folder_identifier_fk
+        try {
+            DbConnectionFactory.getConnection().setAutoCommit(true);
+            if (DbConnectionFactory.isMySql()) {
+                dc.setSQL(MYSQL_DROP_CONSTRAINT_QUERY);
+            } else {
+                dc.setSQL(DROP_CONSTRAINT_QUERY);
+            }
+            Logger.info(this, "Executing Query: " + dc.getSQL());
+            dc.loadResult();
+            DbConnectionFactory.getConnection().setAutoCommit(false);
+        } catch (SQLException e) {
+            Logger.error(this,"Error dropping constraint: " + e.getMessage());
+            throw new DotDataException(e);
+        }
+
+        //Update the Ids
+        updateFolderIDs();
+
+        //Add folder_identifier_fk
+        try {
+            dc = new DotConnect();
+            DbConnectionFactory.getConnection().setAutoCommit(true);
+            dc.setSQL(CREATE_CONSTRAINT_QUERY);
+            Logger.info(this, "Executing Query: " + dc.getSQL());
+            dc.loadResult();
+            DbConnectionFactory.getConnection().setAutoCommit(false);
+        } catch (SQLException e) {
+            Logger.error(this,"Error creating constraint: " + e.getMessage());
+            throw new DotDataException(e);
+        }
+    }
+
+    @WrapInTransaction
+    private void updateFolderIDs() throws DotDataException {
+        //Update identifier table
+        DotConnect dc = new DotConnect();
+        dc.setSQL(UPDATE_IDENTIFIER_QUERY);
+        dc.addParam(SYSTEM_FOLDER_IDENTIFIER);
+        Logger.info(this, "Executing Query: " + dc.getSQL());
+        dc.loadResult();
+
+        //Update folder table
+        dc = new DotConnect();
+        dc.setSQL(UPDATE_FOLDER_QUERY);
+        dc.addParam(SYSTEM_FOLDER_IDENTIFIER);
+        Logger.info(this, "Executing Query: " + dc.getSQL());
+        dc.loadResult();
+    }
+
+    @Override
+    public String getPostgresScript() {
+        return null;
+    }
+
+    @Override
+    public String getMySQLScript() {
+        return null;
+    }
+
+    @Override
+    public String getOracleScript() {
+        return null;
+    }
+
+    @Override
+    public String getMSSQLScript() {
+        return null;
+    }
+
+    @Override
+    public String getH2Script() {
+        return null;
+    }
+
+    @Override
+    protected List<String> getTablesToDropConstraints() {
+        return null;
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Return an empty collection instead of null. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1168)